lib/rspec/matchers/built_in/compound.rb

```diff
@@ -101,6 +123,11 @@ def matcher_supports_block_expectations?(matcher)
           false
         end
 
+        def matcher_list_for(matcher)
```

Comment:
But `BaseMatcher` defines defaul `diffable?` as `false`. I assume user custom matcher does not have to derive from `BaseMatcher`?